roles/deploy/tasks/main.yml

```diff
@@ -100,31 +97,39 @@
     - rake
     - skip_ansible_lint
 
-- debug: msg="{{ table_exists.stdout }}"
-  tags: rake
-- debug: msg="{{ table_exists.stderr }}"
-  tags: rake
-
 #--------------------------
 # Create a rollback release
 
 - name: create a database rollback version
-  command: bash -c "set -o pipefail && pg_dump -h '{{ db_host }}' -U '{{ db_user }}' '{{ db }}' | gzip > '{{ rollback_sql_path }}'"
-  args:
-    creates: "{{ rollback_sql_path }}"
+  postgresql_db:
+    name: "{{ db }}"
+    state: dump
+    target: /tmp/rollback.sql.gz
+  become: yes
+  become_user: postgres
   when: table_exists.stderr.find('does not exist') == -1
 
+- name: copy database dump
+  copy:
+    src: /tmp/rollback.sql.qz
+    dest: "{{ rollback_sql_path }}"
+    remote_src: yes
+
 - name: create a database backup version
   command: cp {{ rollback_sql_path }} {{ releases_path }}/{{ timestamp }}.sql.gz
   tags: skip_ansible_lint
   when: table_exists.stderr.find('does not exist') == -1
 
 - name: create a repo rollback version
-  command: mv {{ current_path }} {{ rollback_path }} removes={{ current_path }}
+  command: mv {{ current_path }} {{ rollback_path }}
+  args:
+    removes: "{{ current_path }}"
   tags: skip_ansible_lint
 
 - name: create a repo backup version
-  command: cp -r {{ rollback_path }} {{ releases_path }}/{{ timestamp }} removes={{ rollback_path }}
+  command: cp -r {{ rollback_path }} {{ releases_path }}/{{ timestamp }}
```

Comment:
I don't really understand the point of `rollback_path`. Isn't enough with a releases folder containing the last N releases and `releases/current` being a symlink pointing to the last one? Then, rolling back is just a matter of making that symlink point to an older release. Not to mention how fast that is would be.